extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/ResteasyStandaloneRecorder.java

```diff
@@ -125,9 +127,16 @@ public void run() {
                 });
             }
         }
-        if (hasClasspathResources) {
+        if (!knownPaths.isEmpty()) {
             router = router == null ? Router.router(vertx) : router;
-            router.route().handler(StaticHandler.create(META_INF_RESOURCES));
+            StaticHandler staticHandler = StaticHandler.create(META_INF_RESOURCES);
+            router.route().handler(ctx -> {
+                if (knownPaths.contains(ctx.request().path())) {
```

Comment:
This will also not serve files that are on the filesystem. Is that an intended behaviour?

Reply:
The way I thought it resources were designed in quarkus was that its all under classpath except in dev mode.  If you look at the code, in dev mode, a different static handler is mounted and put first in the chain.